### PR TITLE
Define CloudWatch Alarm Payload structure

### LINKED ISSUE #51: Is there a cloudwatch events example.
I see the cloudwatch_events.go file in the repo but there is no example for this. I have tried using the type but the import does not seem to work. My problem seems to be getting the details of the message. 

Thanks, I am new to go so it could be my problem, but I think I am using thing correctly.   
My use case is I am trying to use a cloudwatch_event to trigger my lambda.

### PR DESCRIPTION
**Pull request by paritosh16**
# Issue
Tries to solve an issue similar to #51 for CloudWatch Alarm Payload in a SNSEvent. If this this merged, can do another PR to resolve #51 

# Description of changes

## TL;DR

Added struct for CloudWatch Alarm payload. 

## Details

* CloudWatch Alarm information lands up in SNSEntity.
* This payload is a valid json string in the SNSEntity.Message containing information about the alarm that changed state and published to SNS topic triggering the lambda function.
* CloudWatch Alarm payload can contain either:
  * Metrics: A list inside the Trigger section of the payload.
    * Metrics contains the list of metrics alarm is based off. 
  * Metric: Details of the single metric that the alarm is based off.
* A sample event containing Metrics can be found [here](https://github.com/paritosh16/aws-lambda-go/blob/master/events/testdata/cloudwatch-alarm-sns-payload-multiple-metrics.json).
* A sample event containing single Metric information can be found [here](https://github.com/paritosh16/aws-lambda-go/blob/master/events/testdata/cloudwatch-alarm-sns-payload-single-metric.json).

##  Structs Details

* The `CloudWatchAlarmSnsPayload` struct refers to first level in the JSON containing information.
* The `CloudWatchAlarmTrigger` struct refers to the trigger information.
* The `CloudWatchMetricDataQuery` struct refers to the list of Metrics inside Trigger.
* The `CloudWatchMetricStat` struct refers to the MetricStat key inside the Metric Data Query.
* The `CloudWatchMetric` struct refers to the Metric information inside MetricStat.
* The `CloudWatchDimension` struct refers to Dimensions of the metric inside CloudWatchMetric.

## Why so many structs?

* All the structs defined except `CloudWatchAlarmSnsPayload` are already defined in the CloudWatch module in the AWS GO SDK.
* However, the structs in the CloudWatch module don't have any `json` definitions and hence to be in sync with how SNS related structs are defined, these have to defined again.
* Finally, all structs can be defined as nested ones in a single `CloudWatchAlarmSnsPaylaod` but for better readability nested structs are defined separately.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,6 +3,7 @@ module github.com/aws/aws-lambda-go
 go 1.12
 
 require (
-	github.com/stretchr/testify v1.4.0
+	github.com/aws/aws-sdk-go v1.31.12
```
Comment (bmoffatt) [inline go.mod:6]: remove unused dependency

File: events/testdata/cloudwatch-alarm-sns-payload-single-metric.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+    "Records": [
+        {
+            "EventSource": "aws:sns",
+            "EventVersion": "1.0",
+            "EventSubscriptionArn": "arn:aws:sns:us-east-2: 594917905844:Default_CloudWatch_Alarms_Topic_Test: 872f5b40-db58-4161-94af-4fbc7f613012",
```
Comment (bmoffatt) [inline events/testdata/cloudwatch-alarm-sns-payload-single-metric.json:6]: you should go through these examples and sanitize, they contain information about your aws account and resources

File: events/sns.go [hunk 3]
```diff
@@ -30,3 +30,59 @@ type SNSEntity struct {
 	UnsubscribeURL    string                 `json:"UnsubscribeUrl"`
 	Subject           string                 `json:"Subject"`
 }
+
+type CloudWatchAlarmSnsPayload struct {
+	AlarmName        string                 `json:"AlarmName"`
+	AlarmDescription string                 `json:"AlarmDescription"`
+	AWSAccountId     string                 `json:"AWSAccountId"`
```
[the remark this answers is not in the document]
Comment (paritosh16) [reply events/sns.go]: Changed to caps: 29d4ac0dd08c52aa6adfa85250172c75373848e8

File: events/sns.go [hunk 4]
```diff
@@ -30,3 +30,59 @@ type SNSEntity struct {
...
+	NewStateValue    string                 `json:"NewStateValue"`
+	NewStateReason   string                 `json:"NewStateReason"`
+	StateChangeTime  string                 `json:"StateChangeTime"`
+	Region           string                 `json:"Region"`
+	AlarmArn         string                 `json:"AlarmArn"`
```
[the remark this answers is not in the document]
Comment (paritosh16) [reply events/sns.go]: Changed to caps: 29d4ac0dd08c52aa6adfa85250172c75373848e8

File: go.mod (hunk 1 above)
Comment (paritosh16) [reply go.mod]: Removed the unused dependency: e5194c9c68879d2971aa1d107b64025d09bf28df

File: events/testdata/cloudwatch-alarm-sns-payload-single-metric.json (hunk 2 above)
Comment (paritosh16) [reply events/testdata/cloudwatch-alarm-sns-payload-single-metric.json]: Thank you for pointing this out! 

Sanitized the example json payloads: 9f1471ffe8842e044e0261a043db109c233f1d1e

**Comment by bmoffatt** [comment]
Thanks!

**Pull request merged**